enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/state/snapshot/CoreStateDownloaderService.java

```diff
@@ -52,20 +56,38 @@ public CoreStateDownloaderService( JobScheduler jobScheduler, CoreStateDownloade
 
     public synchronized void scheduleDownload( LeaderLocator leaderLocator )
     {
+        if ( stopped )
+        {
+            return;
+        }
+
         if ( currentJob == null || currentJob.hasCompleted() )
         {
             currentJob = new PersistentSnapshotDownloader( leaderLocator, applicationProcess, downloader, log,
                     downloaderPauseStrategy );
-            jobScheduler.schedule( downloadSnapshot, currentJob );
+            jobHandle = jobScheduler.schedule( downloadSnapshot, currentJob );
         }
     }
 
     @Override
     public synchronized void stop() throws Throwable
     {
-        if (currentJob != null)
+        stopped = true;
+
+        if ( currentJob != null )
         {
             currentJob.stop();
         }
+        if ( jobHandle != null )
+        {
+            jobHandle.cancel( false );
```

Comment:
The scheduler is designed to only run one task at the time so this should have no effect. Since it is either currently running or completed.